tests/trace_server/test_calls_query_builder.py

```diff
@@ -2461,3 +2475,696 @@ def test_thread_id_and_turn_id_filter_combined():
             "pb_4": "project",
         },
     )
+
+
+def test_object_ref_filter_simple() -> None:
+    cq = CallsQuery(project_id="project")
+    cq.add_field("id")
+    cq.add_condition(
+        tsi_query.EqOperation.model_validate(
+            {
+                "$eq": [
+                    {"$getField": "inputs.model.temperature"},
+                    {"$literal": 1},
+                ]
+            }
+        )
+    )
+    cq.add_order("started_at", "desc")
+    cq.set_expand_columns(["inputs.model"])
+    assert_sql(
```

Comment:
reading this example, I am starting to imply that your design is:
1. split all filter paths by the expand columns
2. introduce intermediate queries for each "subpath" of the filter.

If this implication is correct, i think there is a feature constraint here that that expand columns in themselves become filters (effectively: must have refs at these paths). For example, given two rows:

```
{key_a: {key_b: val}}
```

and

```
{key_a: REF}
REF = {key_b: val}
```

then:
```
filter $.key_a.key_b == val
```

will match the first row only, and if you specify an expand column of `key_a` then it will only match the second row.

---

I do think this is better than what we have, but admittedly a gap in the most pure sense of the feature.